### PR TITLE
Setup database

### LINKED ISSUE #1: Setup DB

### PR DESCRIPTION
**Pull request by o-bm**
Added fields for solution, challenge and comment. 
close #1

### CODE REVIEW

**Comment by arthurhohinng** [comment]
Previous PR https://github.com/utmgdsc/UML_Mentor/pull/11

File: server/models/Solution.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        description: {
+            type: DataTypes.TEXT,
+            allowNull: false
+        },
+        title: {
+            type: DataTypes.TEXT,
+            allowNull: false
+        },
+        diagram: {
+          type: DataTypes.STRING,
```
Comment (alexapostolu) [inline server/models/Solution.js:26]: Minor thing: Indentation consistency

**Review by alexapostolu — APPROVED** [review]

File: server/models/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+fs
+  .readdirSync(__dirname)
+  .filter(file => {
+    return (
+      file.indexOf('.') !== 0 &&
+      file !== basename &&
+      file.slice(-3) === '.js' &&
+      file.indexOf('.test.js') === -1
```
Comment (Vladmidir) [inline server/models/index.js:26]: what is this code exactly?

File: server/index.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+const PORT = process.env.PORT || 3000;
+
+
+// Sync Sequelize models and create a test user
+db.sequelize.sync({ force: true }).then(async () => { // Use { force: true } cautiously as it will drop existing tables
+    console.log('Database synced');
+
+    // Create a test user
+    try {
```
Comment (Vladmidir) [inline server/index.js:15]: i believe this is for testing purposes, right?

If yes, please mark it as such.

File: server/models/index.js (hunk 2 above)
Comment (o-bm) [reply server/models/index.js:26]: This was generated when I initialized sequelize along with other stuff like the /models/ directory

File: server/index.js (hunk 3 above)
Comment (o-bm) [reply server/index.js:16]: Exactly. 
Done

**Comment by o-bm** [comment]
Committed all changes suggested here.

File: server/models/index.js (hunk 2 above)
Comment (Vladmidir) [reply server/models/index.js:26]: it got auto generated? I didnt know it does that.

File: server/models/index.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+const path = require('path');
+const Sequelize = require('sequelize');
+const process = require('process');
+const basename = path.basename(__filename);
+const env = process.env.NODE_ENV || 'development';
+const config = require(__dirname + '/../config/config.json')[env];
+const db = {};
+
+let sequelize;
+if (config.use_env_variable) {
```
Comment (Vladmidir) [inline server/models/index.js:13]: are we going to need to set up `.env` files to run the server? If yes, please add a note about it in the PR.

Comment (o-bm) [reply server/models/index.js:13]: As of now no but this might be something we would want to do in the future.

**Review by Vladmidir — APPROVED** [review]
looks good to me

**Pull request merged**